lib/widgets/compose_box.dart

```diff
@@ -659,7 +681,14 @@ class _AttachFromCameraButton extends _AttachUploadsButton {
     }
     final length = await result.length();
 
-    return [_File(content: result.openRead(), length: length, filename: result.name)];
+    return [_File(
+      content: result.openRead(),
+      length: length,
+      filename: result.name,
+      mimeType: result.mimeType
+        // TODO pass headerBytes (what's a good way to get that?)
```

Comment:
Looking at what we have on `result`, I think `result.openRead(0, defaultMagicNumbersMaxLength)` should get the data.

And then that's a Stream… poking around that API a bit, one solution is:
```dart
  final headerBytes = await result.openRead(0, defaultMagicNumbersMaxLength).expand((l) => l).toList();
```
(suitably formatted)